admin/app/components/solidus_admin/ui/table/component.rb

```diff
@@ -4,17 +4,65 @@ class SolidusAdmin::UI::Table::Component < SolidusAdmin::BaseComponent
   # @param page [GearedPagination::Page] The pagination page object.
   # @param path [Proc] A callable object that generates the path for pagination links.
   # @param columns [Array<Hash>] The array of column definitions.
-  # @option columns [Symbol] :header The column header.
-  # @option columns [Symbol|Proc] :data The data accessor for the column.
+  # @option columns [Symbol|Proc|String] :header The column header.
+  # @option columns [Symbol|Proc|String] :data The data accessor for the column.
   # @option columns [String] :class_name (optional) The class name for the column.
   # @param pagination_component [Class] The pagination component class (default: component("ui/table/pagination")).
-  def initialize(page:, path: nil, columns: [], pagination_component: component("ui/table/pagination"))
+  def initialize(page:, path: nil, columns: [], batch_actions: [], pagination_component: component("ui/table/pagination"))
```

Comment:
I'm interested in your thoughts about extracting the batch actions into a separate component like @rainerdema did for the pagination. This way, the table component would be less heavy on responsibilities. Do you think that something doable?

What do you think about updating the table preview so it can render dummy batch buttons?

Reply:
I think at some point we'll be able to extract some stuff to internal sub components, but right now I feel it's hard to predict where is best to cut the seam, especially since many things can change with the addition of filtering/sorting/scopes.